icu4c/source/i18n/nfsubs.cpp

```diff
@@ -73,14 +73,15 @@ SameValueSubstitution::~SameValueSubstitution() {}
 
 class MultiplierSubstitution : public NFSubstitution {
     int64_t divisor;
+    const NFRule* owningRule;
 
 public:
     MultiplierSubstitution(int32_t _pos,
         const NFRule *rule,
         const NFRuleSet* _ruleSet,
         const UnicodeString& description,
         UErrorCode& status)
-        : NFSubstitution(_pos, _ruleSet, description, status), divisor(rule->getDivisor())
+        : NFSubstitution(_pos, _ruleSet, description, status), divisor(rule->getDivisor()), owningRule(rule)
```

Comment:
Just making sure, but now that you are saving the pointer to `NFRule*`, can you verify that it does indeed always outlive the `MultiplierSubstitution`? (in C++ this requires manual verification; in Rust it is automatic.) Also, please add docs to this function about the lifetime of the `NFRule*`.

Reply:
The rule owns the substitution; this is just a backpointer.  When the rule is deleted, it'll take the substitution down with it.  See `NFRule::~NFRule()`.